support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/validator/authorization/OAuth20AuthorizationCodeResponseTypeAuthorizationRequestValidator.java

```diff
@@ -66,7 +66,11 @@ public boolean validate(final J2EContext context) {
 
         val clientId = request.getParameter(OAuth20Constants.CLIENT_ID);
         val registeredService = getRegisteredServiceByClientId(clientId);
-
+        if (registeredService == null) {
```

Comment:
This doesn't look right. You will need to let `registeredServiceAccessStrategyEnforcer` do the job, and also make sure the null object isn't used here:

```java
webApplicationServiceServiceFactory.createService(registeredService.getServiceId());
```

So something like this:

```java
val registeredService = getRegisteredServiceByClientId(clientId);
val service = registeredService != null ? webApplicationServiceServiceFactory.createService(registeredService.getServiceId()) : null;
```

...which would then let `registeredServiceAccessStrategyEnforcer` handle all access-related behavior.

Reply:
I made this change but I will look into adding a test for it to see how a missing service is handled.